### PR TITLE
Conect to api

### PR DESCRIPTION
**Pull request by uwadonat**
### Bookstore: connect to API

**In this milestone I implemented followings:**
- Refactor add book and remove book features to persist my changes in the server.
- Use middleware to make async requests to the API.
- Connect application with API and run in the browser.

**Hope to meet all requirements in this milestone**

_Thank you for your time to review my project_

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -20,16 +28,77 @@ export default function books(state = defaultState, action) {
...
+  }
+
+  return async function loadBooksThunk(dispatch) {
+    fetch(
+      `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${APP_ID}/books`,
+    )
+      .then((response) => response.json())
+      .then((json) => dispatch(loadBooks(arrayFormat(json))));
+  };
+}
```
Comment (dasileker) [inline src/redux/books/books.js]: - [x] Let's make a good JS by using ES6 arrow functions. and async and await too

File: src/components/appComponents/booksComponents/AddBooForm.js [hunk 2]
```diff
@@ -23,7 +23,6 @@ function AddBookForm() {
       <h2>Add NEW BOOK</h2>
       <form onSubmit={handleSubmit}>
         <input type="text" name="title" placeholder="Book title" value={formState.title} onChange={handleChange} />
-        <input type="text" name="author" placeholder="Author name" value={formState.author} onChange={handleChange} />
         <select name="category" value={formState.category} onChange={handleChange}>
           <option value="" hidden>Category</option>
           <option value="Sci-Fi">Sci-Fi</option>
```
Comment (dasileker) [inline src/components/appComponents/booksComponents/AddBooForm.js]: - [x] You can create a component for the form and callback the props to make it more clean.

File: src/components/appComponents/booksComponents/bookListComponents/BookItem.js [hunk 3]
```diff
@@ -13,11 +12,11 @@ function BookItem(props) {
 
   function handleClick(e) {
     const payload = Number(e.target.dataset.bookid);
-    dispatch(removeBook(payload));
+    dispatch(removeBookAPI(payload));
   }
 
   return (
-    <div className="row book-item">
+    <div className="book-item">
```
Comment (dasileker) [inline src/components/appComponents/booksComponents/bookListComponents/BookItem.js:19]: - [x] You can use also this in separate component and add the filter to your reducer to make it clean.

File: src/redux/books/books.js [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
 
 const defaultState = [];
 
 export default function books(state = defaultState, action) {
   switch (action.type) {
+    case LOAD_BOOKS:
+      return action.payload.map((bookObject) => ({
+        ...bookObject,
+        author: 'Suzanne Collins',
+        progress: 74,
```
Comment (dasileker) [inline src/redux/books/books.js]: - [x] The keys & values for the **bookObject** ca be placed in the **defaultState** instead.

File: src/redux/books/books.js [hunk 5]
```diff
@@ -1,17 +1,25 @@
...
     case ADD_BOOK:
       return state.concat({
-        id: Date.now(),
+        id: action.payload.item_id,
         title: action.payload.title,
-        author: action.payload.author,
+        author: 'suzanne Collins',
         category: action.payload.category,
-        progress: 0,
+        progress: 74,
```
Comment (dasileker) [inline src/redux/books/books.js:26]: - [x] Also the progress

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @uwadonat** 

Good job so far!
You have done a great job on this Milestone, but let's make this code great see the require changes down in inline comments.♻


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 5 above)
Comment (uwadonat) [reply src/redux/books/books.js:26]: thank you.

Comment (uwadonat) [reply src/redux/books/books.js:26]: requested changes is done.

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉 🎉 

## Hi @uwadonat, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**